Explain sorting in view entities (#9509)
mendix/docs: content/en/docs/refguide/modeling/domain-model/entities/view-entities/use-view-entities.md

The original customer view included address information, but most database optimizers will see that this information is not relevant when counting customers by age, so this information is excluded when retrieving the data. However, the information is still present and can be generated, if requested. 

## Sorting of View Entity Results {#sorting}

> Similar to database views, view entities represent data without enforcing any specific order. As a result, it is not possible to use the `ORDER BY` clause in the view entity OQL to sort query results. However, data retrieved from a view entity can be sorted on retrieval (like regular entities) using options like the [Sorting](/refguide/retrieve/#sorting) feature in a Retrieve microflow activity.

Reviewer: Not sure what you mean by "database views".
Do you mean "As with persistable entities"?
Or even, "As with other types of entity".

Author: I mean SQL views, the feature of relational databases. As this is not a Mendix concept, maybe it requires more details. Many people are confused with why it is not possible to have `ORDER BY` in a view entity, so I am trying to provide as much justification as possible

Author: Reference to that is based on how we define view entiities [here](https://docs.mendix.com/refguide/view-entities/#introduction): `This concept is analogous to views commonly found in relational database systems.`

Reviewer: I still think it is better to describe it here in terms of retrieving lists of objects in Mendix than what happens with SQL? I think just saying in the intro that they are like database views is enough reference to that - people who know about SQL views will know about the ordering issues, people who don't will know it more from the side of what happens with a retrieve action.

So would you be happy with:
```suggestion
Similar to other types of Mendix entity, view entities represent data without enforcing any specific order. As a result, it is not possible to use the `ORDER BY` clause in the view entity OQL to sort query results. However, data retrieved from a view entity can be sorted on retrieval (like regular entities) using options like the [Sorting](/refguide/retrieve/#sorting) feature in a Retrieve microflow activity.
```

Or can you not apply a sort on a retrieve action?

Author: The thing is that people who expect order by to work are exactly the people who come from SQL background, and they expect it to work just as in SQL. That is why the fact that it does not actually work for DB views is a very strong argument IMO. But from the platform perspective, your suggestion makes total sense, so let's go with it

> Similarly to database views, vien entities are representations of data that do not imply any particular order of the results. Due to that, it is not possible to use `ORDER BY` clause in the view entity OQL to sort the results of the query. The outcome of a view entity can be sorted on retrieval. For example, using [Sorting](/refguide/retrieve/#sorting) functionality of a microflow Retrieve activity.

Reviewer 2: Just to be less confusing on what is not allowed, i would reword this to:
`...it is not possible to only use `ORDER BY` clause...`

> Similarly to database views, vien entities are representations of data that do not imply any particular order of the results. Due to that, it is not possible to use `ORDER BY` clause in the view entity OQL to sort the results of the query. The outcome of a view entity can be sorted on retrieval. For example, using [Sorting](/refguide/retrieve/#sorting) functionality of a microflow Retrieve activity.

Reviewer 2: Instead of `The outcome of a view entity` it could be better to say `Data retrieved from a view entity`

> Similarly to database views, vien entities are representations of data that do not imply any particular order of the results. Due to that, it is not possible to use `ORDER BY` clause in the view entity OQL to sort the results of the query. The outcome of a view entity can be sorted on retrieval. For example, using [Sorting](/refguide/retrieve/#sorting) functionality of a microflow Retrieve activity.

Reviewer 2: To make it extra obvious, we can end `can be sorted on retrieval.` with `..., like regular entities`.
